qcodes/instrument/base.py

```diff
@@ -227,7 +227,14 @@ def print_readable_snapshot(self, update=False, max_chars=80):
         for par in sorted(snapshot['parameters']):
             name = snapshot['parameters'][par]['name']
             msg = '{0:<{1}}:'.format(name, par_field_len)
-            val = snapshot['parameters'][par]['value']
+
+            # in case of e.g. ArrayParameters, the parameter may not have
```

Comment:
I would change the docstring to say in case snapshot_value is False otherwise 👍

Reply:
Perhaps a slight preference for `snapshot['parameters'][par].get('value', 'Not available')` in stead of try except

Reply:
I wholeheartedly agree with your viewpoint.